Content.Server/Body/Systems/LungSystem.cs

```diff
@@ -47,7 +47,10 @@ private void OnAddedToBody(EntityUid uid, LungComponent component, AddedToBodyEv
             _popupSystem.PopupEntity(Loc.GetString("lung-behavior-gasp"), uid, Filter.Pvs(uid));
         }
 
-        Inhale(uid, lung.CycleDelay);
+        if (EntityManager.TryGetComponent((uid), out MobStateComponent? mobState) && !mobState.IsCritical())
+        {
+            Inhale(uid, lung.CycleDelay);
+        }
```

Comment:
```suggestion
        if (TryComp(uid, out MobStateComponent? mobState) && !mobState.IsAlive())
            return;

        Inhale(uid, lung.CycleDelay);
```

We have TryComp now that you can use as shorthand.

I also suggest allowing the inhale if the entity doesn't have a MobStateComponent and also making sure that the mob is alive (not dead and not crit).

Reply:
I actually had the isCritical to keep it along the lines of the original commit but I guess this one needs a bit more looking into.

The mob is still breathing whether they gasp or not, it's just that in this scenario gasping also forces inhale.

Inhaling in a crit state means they won't get suffocation damage however.

gasp is only run when taking oxygen damage, should inhale even be called at this point?

Reply:
Yes. The gasp replenishes the mob's required oxygen when they get out of a low oxygen area.

Changing it to IsAlive is just to make it also prevent gasping/inhaling while dead.

Reply:
Thank you for replying so quickly on this one, just trying to make sure I got it right:
With the check in place as isAlive, crit can still breathe normally and live indefinitely not taking oxy damage.
Should this be handled elsewhere or should crit state ignore forced inhale as well?
![image](https://user-images.githubusercontent.com/47410468/146664642-14030b15-08b8-47ff-9cb5-58ac65eeb670.png)

Reply:
Yeah, it's a `!mobState.IsAlive()` (not is alive) check there so it's checking to make sure that it only allows `Inhale` to run if the mob doesn't have a `MobStateComponent` or it has the component and it is alive.

Reply:
Also I'm pretty sure Alive check doesn't count crit as alive. It was a separate state last I checked.

Reply:
Actually I just tested the code with a breakpoint and the `uid` is for the lungs entity. You need the `uid` of the player character.

Reply:
Ok, yeah.. I realize what I did wrong there now. got fooled by lung.owner.

Made changes to use a similar setup to inhale and was able to return as crit, alive still passes.
thanks

![image](https://user-images.githubusercontent.com/47410468/146686975-52d8da19-1152-4865-9bc6-18c0c5cfb350.png)